src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -106,10 +106,21 @@ private static Result<BindingInvocationDescription> GetBindingForGeneration(Gene
 			return Result<BindingInvocationDescription>.Failure(lambdaSymbolResult.Diagnostics);
 		}
 
-		var lambdaTypeInfo = context.SemanticModel.GetTypeInfo(lambdaBodyResult.Value, t);
-		if (lambdaTypeInfo.Type == null)
+		var lambdaParamType = lambdaSymbolResult.Value.Parameters[0].Type;
+		if (lambdaParamType is IErrorTypeSymbol)
 		{
-			return Result<BindingInvocationDescription>.Failure(DiagnosticsFactory.UnableToResolvePath(lambdaBodyResult.Value.GetLocation()));
+			return Result<BindingInvocationDescription>.Failure(DiagnosticsFactory.LambdaParameterTypeCannotBeResolved(lambdaBodyResult.Value.GetLocation()));
+		}
+
+		if (lambdaParamType.DeclaredAccessibility == Accessibility.Private)
```

Comment:
Now that I'm thinking about it, even `protected` and other keyword combinations that are more restrictive than `internal` will cause the issue. I think we should only allow `Public`, `Internal`, or `ProtectedOrInternal` (https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.accessibility?view=roslyn-dotnet-4.9.0) and fail for anything else.